tests/Doctrine/Tests/DBAL/Functional/WriteTest.php

```diff
@@ -334,4 +335,25 @@ public function testDeleteWhereIsNull()
 
         self::assertCount(0, $data);
     }
+
+    /**
+     * Returns the ID of the last inserted row or skips the test if the currently used driver
+     * doesn't support this feature
+     *
+     * @return string
+     *
+     * @throws DriverException
+     */
+    private function lastInsertId(?string $name = null)
+    {
+        try {
+            return $this->connection->lastInsertId($name);
+        } catch (DriverException $e) {
+            if ($e->getCode() === 'IM001') {
```

Comment:
This looks fragile, is it possible it may hide possible bugs in future?

Reply:
This is how PDO [reports](https://github.com/php/php-src/blob/1ad08256f349fa513157437abc4feb245cce03fc/ext/pdo/pdo_sqlstate.c#L283) that it doesn't support a certain feature. I think it's the best we can get from PDO.